File: Subdomain Visit Count.cpp
class Solution {
public:
    vector<string> subdomainVisits(vector<string>& cpdomains) {
        map<string,int> temp;
        for(auto x:cpdomains)
        {
            int y=x.find(" ");
           int n=stoi(x.substr(0,y));
            string str=x.substr(y+1);
            for (int i=0;i<str.length();i++)
            {
                if(str[i]=='.')
                temp[str.substr(i+1)]+=n;

            }
            temp[str]+=n;
        }
            vector <string> ans;
            for (auto z:temp){
                ans.push_back(to_string(z.second)+" " + z.first);
            }
       return ans; 
    }
};

File: week-2/python/subdomain-visit-count.py
class Solution:
    def subdomainVisits(self, cpdomains: List[str]) -> List[str]:
        mp = {}
        ans = []
        for domain in cpdomains:
            n,s = domain.split()
            n=int(n)
            for i in range(len(s)-1,-1,-1):
                if s[i] == ".":
                    subdomain = s[i+1:]
                    if subdomain in mp:
                        mp[subdomain] += n
                    else:
                        mp[subdomain] = n
            subdomain = s[i:]
            if subdomain in mp:
                mp[subdomain] += n
            else:
                mp[subdomain] = n
        for key,value in mp.items():
            ans.append(str(value) + " " + key)  
        return ans        